MdeModulePkg/Core/Pei/Memory/MemoryServices.c

```diff
@@ -636,24 +637,17 @@ PeiAllocatePages (
   //
   // Verify that there is sufficient memory to satisfy the allocation.
   //
-  RemainingPages = (UINTN)(*FreeMemoryTop - *FreeMemoryBottom) >> EFI_PAGE_SHIFT;
+  RemainingMemory = (UINTN)(*FreeMemoryTop - *FreeMemoryBottom);
+  RemainingPages  = RemainingMemory >> EFI_PAGE_SHIFT;

```

Comment:
You might need to use `RShiftU64` here, with the cast, to avoid different compilers inserting intrinsics we don't have (MSVC IIRC)